test/unit/gateways/datatrans_test.rb

```diff
@@ -13,125 +13,148 @@ def setup
       email: 'john.smith@test.com'
     }
 
+    @transaction_reference = '240214093712238757|093712'
+
     @billing_address = address
+
+    @nt_credit_card = network_tokenization_credit_card(
+      '4111111111111111',
+      payment_cryptogram: 'EHuWW9PiBkWvqE5juRwDzAUFBAk=',
+      eci: '07',
+      source: :network_token,
+      verification_value: '737',
+      brand: 'visa'
+    )
+
+    @apple_pay_card = network_tokenization_credit_card(
+      '4900000000000094',
+      payment_cryptogram: 'YwAAAAAABaYcCMX/OhNRQAAAAAA=',
+      month: '06',
+      year: '2025',
+      source: 'apple_pay',
+      verification_value: 569
+    )
   end
 
   def test_authorize_with_credit_card
-    @gateway.expects(:ssl_request).
-      with(
-        :post,
-        'https://api.sandbox.datatrans.com/v1/transactions/authorize',
-        all_of(
-          regexp_matches(%r{"number\":\"(\d+{12})\"}),
-          regexp_matches(%r{"refno\":\"(\d+)\"}),
-          includes('"currency":"CHF"'),
-          includes('"amount":"100"')
-        ),
-        anything
-      ).
-      returns(successful_authorize_response)
-
-    response = @gateway.authorize(@amount, @credit_card, @options)
+    response = stub_comms(@gateway, :ssl_request) do
+      @gateway.authorize(@amount, @credit_card, @options)
+    end.check_request do |_action, endpoint, data, _headers|
+      assert_match('authorize', endpoint)
+      assert_match(/\"number\":\"(\d+{12})\"/, data)
```

Comment:
💬 Comment:

`data` is actually a JSON, why not  just parse it and check the data you need instead of dealing with regex.

Reply:
Oh right, I'll update all of the tests.